### DISCUSSION TITLE
how to get context

### DISCUSSION THREAD

**Discussion by Alma593** [sarchlab/mgpusim#76] [Q&A]
Mr. Author, how do I get the currently running Context  object in the NextWG method in gridbuilder.go ?

**Comment by syifan** [comment]
Can you provide more information about the question? I do not really understand it.

**Comment by bigplanewangyue** [comment]
```go
func (b *gridBuilderImpl) formWavefronts(wg *WorkGroup) {
	var wf *Wavefront
	wavefrontSize := 64
	for i, wi := range wg.WorkItems {
		wg := wi.WG
		inWGID := wi.IDZ*wg.SizeX*wg.SizeY + wi.IDY*wg.SizeX + wi.IDX
		wi.extraFunction(inWGID)
		if inWGID%wavefrontSize == 0 {
			wf = NewWavefront()
			wf.FirstWiFlatID = wg.WorkItems[i].FlattenedID()
			wf.CodeObject = b.hsaco
			wf.Packet = b.packet
			wf.PacketAddress = b.packetAddr
			wf.WG = wg
			wg.Wavefronts = append(wg.Wavefronts, wf)
		}
		wf.WorkItems = append(wf.WorkItems, wi)
		wf.InitExecMask |= 1 << uint32(inWGID%wavefrontSize)
	}
}

type WorkItem struct {
	WG            *WorkGroup
	IDX, IDY, IDZ int
}

func (wi *WorkItem) extraFunction(a int) {
	if a%64 == 63 {
			....
	} 
}
```

I'm trying to periodically add some functions to some workitems in the `formWavefronts` method, and I noticed that the buffer has a field called `l2Dirty`, can I flush the cache by letting `l2Dirty =true`? if I can, how can I  get the `Context` object in `extraFunction` method in `grid.go`?

**Comment by syifan** [comment]
I still do not understand what you want to do here.
